Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,434 @@
+                RemoveEquipment(e.OldItems);
+        }
+
+        private void character_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Character.WoundModifier))
+            {
+                OnPropertyChanged(nameof(WoundModifier));
+            }
+        }
+
+        public int GetTotalMatrixAttribute(string attribute)
+        {
+            switch (attribute)
+            {
+                case "Attack":
+                    return TotalAttack;
+                case "Sleaze":
+                    return TotalSleaze;
+                case "Data Processing":
+                    return TotalDataProcessing;
+                case "Firewall":
+                    return TotalFirewall;
+            }
+            return 0;
+        }
+
+        public int GetTotalSkill(string skillName)
+        {
+            Skill skill = _character.SkillsSection.GetActiveSkill(skillName);
+            if (skill != null)
+                return skill.TotalBaseRating;
+            return 0;
+        }
+
+        public int GetTotalAttribute(string attributeName)
+        {
+            CharacterAttrib attribute = _character.GetAttribute(attributeName);
+            if (attribute != null)
+                return attribute.TotalValue;
+            return 0;
+        }
+
+        /// <summary>
+        /// Parse string as integer or as character attribute by attribute's name
+        /// </summary>
+        /// <param name="value">Integer value or Attribute's name</param>
+        /// <returns></returns>
+        private int Parse(string value)
+        {
+            int result;
+            if (!int.TryParse(value, out result))
+            {
+                CharacterAttrib attribute = _character.GetAttribute(value.Replace("{", "").Replace("}", ""));
+                if (attribute != null)
+                    result = attribute.Base;
+            }
+            return result;
+        }
+
+        #endregion
+
+        #region Properties
+
+        public BindingList<Gear> Persons { get; set; }
+        public BindingList<Program> Programs { get; set; }
+        public List<MatrixAction> Actions { get; set; }
+        public List<Program> ProgramsList { get; }
+
+        public IHasMatrixAttributes CurrentPerson
+        {
+            get => _character.ActiveCommlink;
+            set
+            {
+                if (_character.ActiveCommlink != value)
+                {
+                    _character.ActiveCommlink = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        private int currentActionIndex = 0;
+        public int CurrentActionIndex
+        {
+            get => currentActionIndex;
+            set
+            {
+                currentActionIndex = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public MatrixAction CurrentAction => Actions[currentActionIndex];
```

Comment:
Oof. I see this, and I immediately wince at all the `IndexOutOfRange` exceptions that are going to happen. You can do one of two things. First option, you need to change this to include safety checks so that `currentActionIndex` (the member, not the property, because you are using the member here and not the property) is always in range of `Actions`. Second option, you need to set up some maintenance code that makes sure any time `currentActionIndex` and/or `Actions` is changed (remember, you are using `currentActionIndex` the member here, not the property) so that the former is always in range of the latter. Either way, this code *needs* to be fixed.